### PR TITLE
add image file name validation.

### PR DESCRIPTION
**Pull request by thefrieddan1**
add image file name validation.

## Related Issues
fixes: https://jira-dc.paloaltonetworks.com/browse/CIAC-5180

## Description
Add image file name validation to validate_content_path script

### CODE REVIEW

File: demisto_sdk/commands/common/constants.py [hunk 1]
```diff
@@ -1233,14 +1233,17 @@ class FileType(str, Enum):
...
-TESTS_AND_DOC_DIRECTORIES = [
+TESTS_DIRECTORIES = [
     "testdata",
     "test_data",
     "data_test",
     "tests_data",
+    "TestData",
+]
+
+DOCS_DIRECTORIES = [
```
[the remark this answers is not in the document]
Comment (thefrieddan1) [reply demisto_sdk/commands/common/constants.py:1244]: Nice! (and smart)

File: demisto_sdk/scripts/validate_content_path.py [hunk 2]
```diff
@@ -295,6 +312,12 @@ def _validate(path: Path) -> None:
             raise InvalidXDRCTemplatesFileName
 
 
+def _validate_image_file_name(image_name: str):
+    pattern = r"[^0-9a-zA-Z-_]+"
+    if re.findall(pattern, image_name):
```
[the remark this answers is not in the document]
Comment (thefrieddan1) [reply demisto_sdk/scripts/validate_content_path.py:313]: fullmatch does not catch the invalid img file names from the unit test case

File: demisto_sdk/scripts/validate_content_path.py [hunk 3]
```diff
@@ -283,6 +290,9 @@ def _validate(path: Path) -> None:
         ):
             raise InvalidXSIAMReportFileName
 
+        if first_level_folder == DOC_FILES_DIR:
```
Comment (dorschw) [inline demisto_sdk/scripts/validate_content_path.py:293]:
```suggestion
        if first_level_folder == DOC_FILES_DIR and path.suffix in (".png", ".jpg", ".jpeg", ".gif"):
```
1. Have I missed any suffixes?
2. Looks like we call it on all files under `DOC_FILES_DIR`. Please insert this change and then modify the unit tests to match (make sure that it only fails on image files, test it with image suffixes and with non-image and make sure non-image files can contain chars that images are not allowed to).

File: demisto_sdk/commands/common/legacy_git_tools.py [hunk 4]
```diff
@@ -346,7 +346,7 @@ def filter_changed_files(
                 continue
 
             # ignore changes in TESTS_DIRECTORIES files.
-            elif any(test_dir in file_path for test_dir in TESTS_AND_DOC_DIRECTORIES):
+            elif any(test_dir in file_path for test_dir in (TESTS_AND_DOC_DIRECTORIES)):
```
Comment (dorschw) [inline demisto_sdk/commands/common/legacy_git_tools.py:349]:
```suggestion
            elif any(test_dir in file_path for test_dir in TESTS_AND_DOC_DIRECTORIES):
```

File: demisto_sdk/scripts/validate_content_path.py [hunk 5]
```diff
@@ -127,6 +129,7 @@
     )
 )
 DIRS_ALLOWING_SPACE_IN_FILENAMES = (TEST_PLAYBOOKS_DIR,)
+IMAGE_NAME_REGEX = re.compile(r"[^0-9a-zA-Z-_]+")
```
Comment (dorschw) [inline demisto_sdk/scripts/validate_content_path.py:132]:
```suggestion
INVALID_CHARS_IN_IMAGES_REGEX = re.compile(r"[^0-9a-zA-Z-_]+")
```

File: demisto_sdk/scripts/validate_content_path.py (hunk 3 above)
Comment (thefrieddan1) [reply demisto_sdk/scripts/validate_content_path.py:293]: The constant ALLOWED_SUFFIXES support only jpg and svg

**Review by dorschw — APPROVED** [review]

**Pull request merged**